### PR TITLE
:shipit: Assignment 4 - Enums

### PR DESCRIPTION
**Pull request by ray2g**
This PR adds a new module:

 - `region_enums`: A Enum class which defines a list of possible regions to filter the data from.

Additional a test were created to this module,

### CODE REVIEW

File: assignments/life_expectancy/cleaning.py [hunk 1]
```diff
@@ -52,6 +53,8 @@ def clean_data(df: DataFrame, region: str) -> DataFrame:
     df_clean = df_unpivot.dropna()
 
     # filter dataframe by desire region
-    df_clean_filt = df_clean[df_clean['region'] == region]
+    df_clean_filt = df_clean[df_clean['region'] == region.value]
```
Comment (jpleitao) [inline assignments/life_expectancy/cleaning.py:57]: Minor suggestion (conditioned on interpretation of the assignment): I considered matching the region in both upper and lower case (e.g., if the input df contains one row with region "PT" and another with "pt", both would be selected). Not sure if this is the goal of the assignment or not; anyway it is a minor point

**Review by jpleitao — APPROVED** [review]

File: assignments/life_expectancy/main.py [hunk 2]
```diff
@@ -47,7 +47,8 @@ def main(i_path, region, o_path): # pragma: no cover
     parser = argparse.ArgumentParser()
     parser.add_argument('-i', help= "Input file path.") 
     parser.add_argument('-o', help = "Output file path.")
-    parser.add_argument("-r", default="PT", help = "Desire region to filter the csv.")
+    parser.add_argument("-r",  type = Region, choices = list(Region), \
```
Comment (MrCordeiro) [inline assignments/life_expectancy/main.py:50]:
```suggestion
    parser.add_argument("-r",  type = Region, choices = Region, \
```

You don't really need the `list` here

I don't you need the list here

Comment (MrCordeiro) [inline assignments/life_expectancy/main.py:50]: Something that's worth pointing it out is that this parser will accept `main.py -r Region.PT` but it won't accept `main.py -r PT`.

If you want to accept the later, you could create a custom class method, like this:

```python
import enum
import argparse

class MyEnum(enum.Enum):
    A = 1
    B = 2
    C = 3

    @classmethod
    def argtype(cls, s: str):
        try:
            return cls[s]
        except KeyError:
            raise argparse.ArgumentTypeError(
                f"{s!r} is not a valid {cls.__name__}")

parser = argparse.ArgumentParser()
parser.add_argument('--enum', type=MyEnum.argtype , choices=MyEnum, required=True)
```

File: assignments/life_expectancy/cleaning.py [hunk 3]
```diff
@@ -52,6 +53,8 @@ def clean_data(df: DataFrame, region: str) -> DataFrame:
...
 
     return df_clean_filt
+
+# %%
```
Comment (MrCordeiro) [inline assignments/life_expectancy/cleaning.py:60]:
```suggestion
```

**Review by MrCordeiro — APPROVED** [review]
Everything looks solid! Just game a few suggestions.

File: assignments/life_expectancy/regions_enum.py
[the remark this answers is not in the document]
Comment (ray2g) [reply assignments/life_expectancy/regions_enum.py:1]: Changed!

File: assignments/life_expectancy/tests/test_regions_enum.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    :param expected_regions: Fixture which is a list of valid regions.
+    """
+
+    regions = Region.considered_regions()
+
+    # Check if both lists of regions have the same number of elements
+    assert len(regions) == len(expected_regions)
+    
+    # Check if both lists of regions have the same elements
+    assert not set(regions) ^ set(expected_regions)
```
[the remark this answers is not in the document]
Comment (ray2g) [reply assignments/life_expectancy/tests/test_regions_enum.py]: Thanks! Changed

File: assignments/life_expectancy/main.py [hunk 5]
```diff
@@ -56,4 +57,5 @@ def main(i_path, region, o_path): # pragma: no cover
         if arg not in args:
             raise ValueError(f"Missing argument: '{arg}'")
```
[the remark this answers is not in the document]
Comment (ray2g) [reply assignments/life_expectancy/main.py:59]: Changed

File: assignments/life_expectancy/main.py (hunk 2 above)
Comment (ray2g) [reply assignments/life_expectancy/main.py:50]: Changed

Comment (ray2g) [reply assignments/life_expectancy/main.py]: Thanks, will take that in consideration

**Pull request merged**